feat(backend/dtr): suppress dtr asset and contract creation via parameter

tom-rm-meyer-ISST opened the pull request:
<!-- 
Thanks for your contribution! 
Please follow the instructions on your PRs title and description.
aligned title description: '(feat|fix|chore|doc): _description of introduced change_'
Important: Contributing Guidelines can be found here: https://eclipse-tractusx.github.io/docs/oss/how-to-contribute
Info: <!- text comments ->  will be hidden from the rendered preview of your PR.
-->

## Description

solves #964 

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files
- [x] If helm chart has been changed, the chart version has been bumped to either next major, minor or patch level (compared to released chart).

ReneSchroederLJ commented on local/deploy.sh:
```diff
@@ -110,14 +110,37 @@ docker compose down -v
+  printf '.'
+  sleep 5
+  done
+  npm run local-test-prepare-existing-asset
```
This should be `npm run local-test-prepare-existing-assets` to match the defined script in the `package.json`

ReneSchroederLJ commented on local/postman/README.md:
```diff
@@ -12,6 +12,17 @@ It creates a setup of a partner and customer that refer to the same material fro
+
+We provide the following test scenarios as mentioned in package.json:
+
+- EDC and DTR are shared (currently only shared DTR asset) -> `npm run <env>-test-prepare-existing-asset`
```
Same as mentioned in `deploy.sh` the name of the script is using the plural assets. Either change the script name or the references to it.

ReneSchroederLJ commented on local/bruno/puris-integration-test/environments/local.bru:
```diff
@@ -1,10 +1,10 @@
+  CUSTOMER_MANAGE_CLIENT_SECRET:
+  SUPPLIER_EDC_API_KEY:
+  SUPPLIER_PURIS_BACKEND_API_KEY:
+  SUPPLIER_MANAGE_CLIENT_SECRET:
```
The removal of these spaces breaks key generation for me. Without the spaces the matching within the `generate-keys.sh` script no longer works and no keys are added.

ReneSchroederLJ reviewed (changes requested):
Thank you for your great contribution. Very in-depth and and useful. I noticed some bugs that need to be addressed and I ran into an issue with the bru CLI.

Regarding the use of the bru CLI:

At the moment, despite having a local NPM environment for the integration tests, we use the globally installed version of the bru CLI for running the integration tests. 

In practice this leads to 2 issues:

- the bru CLI needs to be installed globally to run the integration tests
- a certain minimum version of the CLI is required to get the expected results

The latter issue came up for me during review. I was running version 2.1.0 of the CLI, which does not support all flags used in the new bru run command. I updated to the latest version 2.10.1 to fix the issue, but it would be desirable to provide guidance for users of the repo on how to use the integration tests.

I would suggest we either document the need for a global installation of the CLI including a minimum version or even better update the versions listed in the `package.json` and switch to using a local installation of the CLI if possible.

tom-rm-meyer-ISST commented:
> Thank you for your great contribution. Very in-depth and useful. I noticed some bugs that need to be addressed and I ran into an issue with the bru CLI.
> 
> Regarding the use of the bru CLI:
> 
> At the moment, despite having a local NPM environment for the integration tests, we use the globally installed version of the bru CLI for running the integration tests.
> 
> In practice this leads to 2 issues:
> 
> * the bru CLI needs to be installed globally to run the integration tests
> * a certain minimum version of the CLI is required to get the expected results
> 
> The latter issue came up for me during review. I was running version 2.1.0 of the CLI, which does not support all flags used in the new bru run command. I updated to the latest version 2.10.1 to fix the issue, but it would be desirable to provide guidance for users of the repo on how to use the integration tests.
> 
> I would suggest we either document the need for a global installation of the CLI including a minimum version or even better update the versions listed in the `package.json` and switch to using a local installation of the CLI if possible.

Fair, updated documentation and bumped the version. I requested IP clarification on several items, thus the IP check failes:

 [main] INFO This content is either not correctly mapped by the system, or requires review.
[main] INFO A review is required for npm/npmjs/@smithy/node-http-handler/4.2.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23206 .
[main] INFO A review is required for npm/npmjs/@smithy/smithy-client/4.6.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23207 .
[main] INFO A review is required for npm/npmjs/@smithy/util-config-provider/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23208 .
[main] INFO A review is required for npm/npmjs/@smithy/util-defaults-mode-node/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23209 .
[main] INFO A review is required for npm/npmjs/@usebruno/js/0.37.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23210 .
[main] INFO A review is required for npm/npmjs/@usebruno/lang/0.28.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23211 .
[main] INFO A review is required for npm/npmjs/@smithy/node-config-provider/4.2.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23212 .
[main] INFO A review is required for npm/npmjs/@smithy/util-defaults-mode-browser/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23213 .
[main] INFO A review is required for npm/npmjs/@smithy/types/4.4.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23214 .
[main] INFO A review is required for npm/npmjs/@smithy/invalid-dependency/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23215 .
[main] INFO A review is required for npm/npmjs/@smithy/util-uri-escape/4.1.0.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23216 .
[main] INFO A review is required for npm/npmjs/@smithy/util-utf8/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23217 .
[main] INFO A review is required for npm/npmjs/@smithy/middleware-content-length/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23218 .
[main] INFO A review is required for npm/npmjs/@usebruno/schema/0.18.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23219 .
[main] INFO A review is required for npm/npmjs/@usebruno/crypto-js/3.1.9.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23220 .
[main] INFO A review is required for npm/npmjs/@smithy/util-body-length-node/4.1.0.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23221 .
[main] INFO A review is required for npm/npmjs/@smithy/querystring-builder/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23222 .
[main] INFO A review is required for npm/npmjs/@smithy/config-resolver/4.2.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23223 .
[main] INFO A review is required for npm/npmjs/@smithy/is-array-buffer/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23224 .
[main] INFO A review is required for npm/npmjs/@smithy/middleware-endpoint/4.2.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23225 .
[main] INFO A review is required for npm/npmjs/@usebruno/filestore/0.2.0.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23226 .
[main] INFO A review is required for npm/npmjs/@smithy/fetch-http-handler/5.2.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23227 .
[main] INFO A review is required for npm/npmjs/@smithy/abort-controller/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23228 .
[main] INFO A review is required for npm/npmjs/@smithy/shared-ini-file-loader/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23229 .
[main] INFO A review is required for npm/npmjs/@smithy/credential-provider-imds/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23230 .
[main] INFO A review is required for npm/npmjs/@smithy/util-middleware/4.1.0.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23231 .
[main] INFO A review is required for npm/npmjs/-/flow-parser/0.281.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23232 .
[main] INFO A review is required for npm/npmjs/@smithy/url-parser/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23233 .
[main] INFO A review is required for npm/npmjs/-/quickjs-emscripten-core/0.29.2.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23234 .
[main] INFO A review is required for npm/npmjs/@usebruno/common/0.11.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23235 .
[main] INFO A review is required for npm/npmjs/@smithy/signature-v4/5.2.0.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23236 .
[main] INFO A review is required for npm/npmjs/@smithy/util-body-length-browser/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23237 .
[main] INFO A review is required for npm/npmjs/@smithy/middleware-serde/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23238 .
[main] INFO A review is required for npm/npmjs/@smithy/property-provider/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23239 .
[main] INFO A review is required for npm/npmjs/@smithy/util-base64/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23240 .
[main] INFO A review is required for npm/npmjs/@smithy/core/3.10.0.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23241 .
[main] INFO A review is required for npm/npmjs/-/js-md4/0.3.2.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23242 .
[main] INFO A review is required for npm/npmjs/@usebruno/requests/0.8.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23243 .
[main] INFO A review is required for npm/npmjs/@smithy/middleware-retry/4.2.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23244 .
[main] INFO A review is required for npm/npmjs/@smithy/protocol-http/5.2.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23245 .
[main] INFO A review is required for npm/npmjs/@usebruno/cli/2.10.1.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23246 .
[main] INFO A review is required for npm/npmjs/@smithy/util-retry/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23247 .
[main] INFO A review is required for npm/npmjs/@smithy/util-stream/4.3.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23248 .
[main] INFO A review is required for npm/npmjs/-/postman-request/2.88.1-postman.43.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23249 .
[main] INFO A review is required for npm/npmjs/@usebruno/converters/0.11.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23250 .
[main] INFO A review is required for npm/npmjs/@smithy/middleware-stack/4.1.0.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23251 .
[main] INFO A review is required for npm/npmjs/-/jscodeshift/17.3.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23252 .
[main] INFO A review is required for npm/npmjs/@smithy/util-buffer-from/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23253 .
[main] INFO A review is required for npm/npmjs/-/grpc-reflection-js/0.3.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23254 .
[main] INFO A review is required for npm/npmjs/@smithy/util-hex-encoding/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23255 .
[main] INFO A review is required for npm/npmjs/@smithy/util-endpoints/3.1.0.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO GitLab API rate limit reached. Will retry in 30 seconds.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23256 .
[main] INFO A review is required for npm/npmjs/@smithy/service-error-classification/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23257 .
[main] INFO A review is required for npm/npmjs/@smithy/querystring-parser/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23258 .
[main] INFO A review is required for npm/npmjs/@smithy/hash-node/4.1.0.
[main] INFO A review request was created https://gitlab.eclipse.org/eclipsefdn/emo-team/iplab/-/issues/23259 .

mhellmeier reviewed (approved):
LGTM. The dependency checks are okay now. The >1000 LOC are mostly related to the nearly 10k  `package-lock.json` file. It it sill okay since the PR owner is a Committer (see TRG 7.00).
